### ISSUE TITLE
Desktop (Linux, MacOS, Windows) and Android/iOS application?

### ISSUE THREAD

**Issue by RaptaG** [snuffyDev/Beatbump#122]
Just a suggestion

**Comment by IronMaltese** [comment]
Is there a way to do this with Electron? Since Beatbump is a web app

**Comment by BrightDV** [comment]
For android, there is a FOSS alternative: [see here (comment)](https://github.com/snuffyDev/Beatbump/issues/59#issuecomment-1242216925)

**Comment by IronMaltese** [comment]
Here’s a video on how to do this with electron 

https://odysee.com/@DistroTube:2/how-to-turn-any-website-into-a-desktop:b?r=BmnD2fSvimXii1DZ7PNAM5WnhVYmzvwV

**Comment by snuffyDev** [comment]
I've been experimenting/thinking about this - except with Tauri instead of Electron. I really like working with Rust, and the benefits for using Tauri are very in-line with this project's goals (lightweight, fast, simple) 

Another app of mine (Harmony, WIP postman alternative) is only <4mb installed https://github.com/snuffyDev/Harmony

Also @IronMaltese, I've been gone awhile (don't worry, I'm okay!) since I was taking a break from Beatbump to "refresh". I appreciate you greatly for keeping Issues active and for responding to people! I've managed to land an internship (my first tech job!), but my bosses know Beatbump (its the reason they hired me) so they know I'm gonna be working on this!

Keep an eye out for a commit/release very soon! I'm updating SvelteKit and the adapters, as well as hopefully releasing some notes for contributors/instance admins/etc
![image](https://user-images.githubusercontent.com/72365477/202939238-ef330e7c-c4da-45e0-9ee4-28a99e0c55c0.png)

**Comment by IronMaltese** [comment]
It’s great to see you back! I knew you said you were looking for a job, so I wasn’t worried, I’m glad you took a break instead of tiring yourself out with such a busy schedule :)

**Comment by IntoTheVoid-900** [comment]
Great to see you back! None of the other desktop options quite work for me (Spotube doesn't like my firewall, nuclear is... nuclear, I don't have the battery for WSA or Bluestacks to run an Android app), so I use Beatbump as a PWA daily. Honestly, I think the PWA replaces the need for a Tauri desktop app. ViMusic does the job on Android.

**Comment by IronMaltese** [comment]
> None of the other desktop options quite work for me (Spotube doesn't like my firewall, nuclear is... nuclear, I don't have the battery for WSA or Bluestacks to run an Android app), so I use Beatbump as a PWA daily.

I just wanted to share my experience, we all know that Nuclear is a different beast of an application entirely, so I've only tinkered around with it for a week or so and I never heard of spotube, but it requiring a spotify account is a dealbreaker. Looking at Hyperpipe, it really has great potential but the interface and reliability hold it back, I don't doubt that these things will be fixed in the future though. Beatbump seems to be the best option right now for simple music listening

> Honestly, I think the PWA replaces the need for a Tauri desktop app. ViMusic does the job on Android.

Speaking from personal experience, I know people that didn't want to use beatbump because they would rather control their music in an app, and they didn't know what a web-app was, so this feature would definitely help people who are used to apps and want to try beatbump

**Comment by osfanbuff63** [comment]
I've been working with this, ironically also on Tauri, but just found this issue. It seems to compile fine with the couple needed changes to use Tauri (adapter-static related) but otherwise I haven't changed anything. It doesn't quite work yet; but I just need to do a bit more debugging. Last time I checked it worked in dev mode (e.g. `yarn dev`) though/

---

On mobile, Tauri is working on support for those devices - but it's still in its first alpha release so it might not be worth adopting at this point in time.

**Comment by techgamers2** [comment]
just install (ungoogled chromium ) go to its settings > performance > check smart memory / discord tab option . search beatbump github select instances url closer to ur area https://github.com/snuffyDev/Beatbump#instances > install and app option will pop up in address bar > click install 
now u will see new icon on desktop name beatbump Simple open it and see beatbump lite version only consume <70mb ram better than electron

**Comment by osfanbuff63** [comment]
I think it's important to note that this isn't the same thing we're talking about here - it's a PWA (Progressive Web Application), so it's essentially a shortcut to open Beatbump in that browser. It's a solution in the meantime though.

**Comment by techgamers2** [comment]
web apps r the future ... i switched from youtube music desktop app to beatbump lite web app which performs way better than full fledged desktop app very lite on sys resources so imo desktop app for beatbump is not necessary when web app gets tht job done and it not just a shortcut to the browser when u use ungoogled chromium not whole browser gets loaded into memory thts y i mentioned to enable performance tweaks memory settings to make it feel like standalone app

**Comment by IntoTheVoid-900** [comment]
> a PWA (Progressive Web Application), so it's essentially a shortcut to open Beatbump in that browser.
@osfanbuff63 

Not necessarily. A PWA is much more than just a browser shortcut, it is much faster and lighter than a regular browser tab, and more system integrated. Some PWAs even work offline

**Comment by osfanbuff63** [comment]
True, I guess the word I was looking for was a native app? IDK

**Comment by IntoTheVoid-900** [comment]
The thing is, an electron app would be slower than the PWA, and Tauri would only be marginally better. I am not sure if it is worth the effort when it already works as a PWA
